independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathTree.java

```diff
@@ -1,21 +1,39 @@
 package io.quarkus.paths;
 
+import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
 import java.util.Collection;
 import java.util.function.Function;
 import java.util.jar.Manifest;
 
 public interface PathTree {
 
     static PathTree of(Path p) {
-        if (Files.isDirectory(p)) {
-            return new DirectoryPathTree(p);
-        }
-        if (Files.exists(p)) {
+        try {
+            BasicFileAttributes fileAttributes = Files.readAttributes(p, BasicFileAttributes.class);
+            if (fileAttributes.isDirectory()) {
+                return new DirectoryPathTree(p);
+            }
+
             return new FilePathTree(p);
+        } catch (IOException e) {
+            throw new IllegalArgumentException(p + " does not exist");
+        }
+    }
+
+    static PathTree ofMultiContent(Path p) {
```

Comment:
Any ideas for a better name? was not sure about this one.

Reply:
I guess it could named after its assumptions, e.g. `ofDirectoryOrArchive()`

Reply:
Good idea!